platform/linuxbsd/display_server_x11.cpp

```diff
@@ -829,11 +828,17 @@ void DisplayServerX11::window_set_current_screen(int p_screen, WindowID p_window
 		Point2i position = screen_get_position(p_screen);
 		Size2i size = screen_get_size(p_screen);
 
+		WindowData &wd = windows[p_window];
 		XMoveResizeWindow(x11_display, wd.x11_window, position.x, position.y, size.x, size.y);
 	} else {
-		if (p_screen != window_get_current_screen(p_window)) {
-			Point2i position = screen_get_position(p_screen);
-			XMoveWindow(x11_display, wd.x11_window, position.x, position.y);
+		int current_screen = window_get_current_screen(p_window);
+		if (p_screen != current_screen) {
+
+			Point2i to_screen_pos = screen_get_position(p_screen);
```

Comment:
Notes for the future improvemet:
* Special Fullscreen case probably should be ported to Windows and macOS as well.
* In case current window position (relative to the current screen) is bigger than target screen size, window will be moved to the wrong screen (not handled on any platform).

e.g. with the following display config, moving window to screen 1 will move it to screen 2 instead: <img width="384" alt="Screenshot 2020-04-13 at 12 14 49" src="https://user-images.githubusercontent.com/7645683/79109135-6d399e80-7d80-11ea-9200-963a6d64e96f.png">